crates/subspace-service/src/dsn_archiver.rs

```diff
@@ -0,0 +1,55 @@
+use futures::StreamExt;
+use sc_consensus_subspace::{ArchivedSegmentNotification, SubspaceLink};
+use sp_core::traits::SpawnEssentialNamed;
+use sp_core::Encode;
+use sp_runtime::traits::Block as BlockT;
+use subspace_networking::PUB_SUB_ARCHIVING_TOPIC;
+use tracing::{error, info, trace};
+
+/// Start an archiver that will listen for archived segments and send it to DSN network using
+/// pub-sub protocol.
+pub fn start_subspace_dsn_archiver<Block>(
+    subspace_link: &SubspaceLink<Block>,
+    node_config: subspace_networking::Config,
+    spawner: &impl SpawnEssentialNamed,
+) where
+    Block: BlockT,
+{
+    spawner.spawn_essential_blocking(
```

Comment:
Does it need to be blocking though? Looks like there isn't anything heavy here, so we can use regular spawning without occupying a separate thread.